sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -480,6 +487,82 @@ def test_with_input_types_decorator_violation(self):
         pc = p | Create(l_3_tuple)
         _ = pc | beam.CombineGlobally(self.fn)
 
+#
+# Test cases for streaming.
+#
+@attr('ValidatesRunner')
+class TimestampCombinerTest(unittest.TestCase):
+
+  class RecordFn(beam.DoFn):
+    def process(self, elm=beam.DoFn.ElementParam, ts=beam.DoFn.TimestampParam):
+      yield (elm, ts)
+
+  @unittest.skip('BEAM-8657')
+  def test_combiner_earliest(self):
+    """Test TimestampCombiner with EARLIEST."""
+    options = PipelineOptions()
+    options.view_as(StandardOptions).streaming = True
+    p = TestPipeline(options=options)
+
+    main_stream = (p
+                   | 'main TestStream' >> TestStream()
+                   .add_elements([window.TimestampedValue(('k', 100), 0)])
+                   .add_elements([window.TimestampedValue(('k', 400), 9)])
+                   .advance_watermark_to_infinity()
+                   | 'main windowInto' >> beam.WindowInto(
+                       window.FixedWindows(10),
+                       timestamp_combiner=TimestampCombiner.OUTPUT_AT_EARLIEST)
+                   | 'Combine' >> beam.CombinePerKey(sum))
+
+    records = (main_stream | beam.ParDo(self.RecordFn()))
+
+    # All the KV pairs are applied GBK using EARLIEST timestamp.
+    expected_window_to_elements = {
+        window.IntervalWindow(0, 10): [
+            (('k', 500), Timestamp(0)),
+        ],
+    }
+
+    assert_that(
+        records,
+        equal_to_per_window(expected_window_to_elements),
+        use_global_window=False,
+        label='assert per window')
+
+    p.run()
+
+  def test_combiner_latest(self):
+    """Test TimestampCombiner with LATEST."""
+    options = PipelineOptions()
+    options.view_as(StandardOptions).streaming = True
+    p = TestPipeline(options=options)
+
+    main_stream = (p
+                   | 'main TestStream' >> TestStream()
+                   .add_elements([window.TimestampedValue(('k', 100), 0)])
+                   .add_elements([window.TimestampedValue(('k', 400), 9)])
+                   .advance_watermark_to_infinity()
+                   | 'main windowInto' >> beam.WindowInto(
+                       window.FixedWindows(10),
+                       timestamp_combiner=TimestampCombiner.OUTPUT_AT_EARLIEST)
```

Comment:
Shouldn't this be LATEST?

Reply:
oops.  result from OUTPUT_AT_EARLIEST is indeed wrong, which leads to the false positive here. 

Fixed.